File: C-Sharp/57. Insert Interval.cs
using System.Diagnostics.CodeAnalysis;

namespace LeetCode.C_Sharp;

[SuppressMessage("ReSharper", "UnusedType.Global")]
[SuppressMessage("ReSharper", "UnusedMember.Global")]
public partial class Solution
{
    public int[][] Insert(int[][] intervals, int[] newInterval)
    {
        List<int[]> result = new List<int[]>();

        for (int i = 0; i < intervals.Length; i++)
        {
            int[] current = intervals[i];

            if (newInterval.Length == 0 || current[1] < newInterval[0])
            {
                result.Add(current);
            }
            else if (current[0] > newInterval[1])
            {
                result.Add(newInterval);
                result.Add(current);
                newInterval = [];
            }
            else
            {
                newInterval = new int[] { Math.Min(newInterval[0], current[0]), Math.Max(newInterval[1], current[1]) };
            }
        }

        if (newInterval.Length > 0)
        {
            result.Add(newInterval);
        }

        return result.ToArray();
    }
}

File: JavaScript/57. Insert Interval.js
// noinspection JSUnusedGlobalSymbols

/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
const insert = (intervals, newInterval) => {
  const result = []

  for (let i = 0; i < intervals.length; i++) {
    const current = intervals[i]

    if (newInterval.length === 0 || current[1] < newInterval[0]) {
      result.push(current)
    } else if (current[0] > newInterval[1]) {
      result.push(newInterval)
      result.push(current)
      newInterval = []
    } else {
      const start = Math.min(newInterval[0], current[0])
      const end = Math.max(newInterval[1], current[1])
      newInterval = [start, end]
    }
  }

  if (newInterval.length !== 0) {
    result.push(newInterval)
  }
  return result
}
